cmd/show/dashboard.go

```diff
@@ -0,0 +1,323 @@
+package show
+
+import (
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
+	"sync"
+
+	"github.com/charmbracelet/bubbles/table"
+	tea "github.com/charmbracelet/bubbletea"
+	"github.com/charmbracelet/lipgloss"
+	client "github.com/open-sauced/go-api/client"
+)
+
+const (
+	newContributorsView = iota
+	alumniContributorsView
+)
+
+// DashboardModel holds all the information related to the repository queried (issues, stars, new contributors, alumni contributors)
+type DashboardModel struct {
+	newContributorsTable    table.Model
+	alumniContributorsTable table.Model
+	RepositoryInfo          *client.DbRepo
+	tableView               int
+}
+
+// SelectMsg: message to signal the main model that we want to go to the contributor model when 'enter' is pressed
+type SelectMsg struct {
+	contributorName string
+}
+
+// InitDashboard: initializes the dashboard model
+func InitDashboard(opts *Options) (DashboardModel, error) {
+	model := DashboardModel{}
+	err := validateShowQuery(opts)
+	if err != nil {
+		return model, err
+	}
+	ownerRepo := strings.Split(opts.RepoName, "/")
+
+	resp, r, err := opts.APIClient.RepositoryServiceAPI.FindOneByOwnerAndRepo(opts.ServerContext, ownerRepo[0], ownerRepo[1]).Execute()
+	if err != nil {
+		return model, err
+	}
+
+	if r.StatusCode != 200 {
+		return model, fmt.Errorf("HTTP status: %d", r.StatusCode)
+	}
+
+	model.RepositoryInfo = resp
+
+	var (
+		errorChan = make(chan error)
+		doneChan  = make(chan bool)
+		wg        sync.WaitGroup
+	)
+
+	wg.Add(2)
+	go func() {
+		defer wg.Done()
+		newContributors, err := FetchNewContributors(opts, resp.Id)
+		if err != nil {
+			errorChan <- err
+			return
+		}
+		model.newContributorsTable = setupContributorsTable(newContributors)
+	}()
+
+	go func() {
+		defer wg.Done()
+		alumniContributors, err := FetchAlumniContributors(opts, resp.Id)
+		if err != nil {
+			errorChan <- err
+			return
+		}
+		model.alumniContributorsTable = setupContributorsTable(alumniContributors)
+	}()
+
+	go func() {
+		wg.Wait()
+		close(doneChan)
+	}()
+
+	select {
+	case <-doneChan:
+		break
+	case err := <-errorChan:
+		close(errorChan)
+		return model, err
+	}
+
+	return model, nil
+}
+
+func (m DashboardModel) Init() tea.Cmd {
+	return nil
+}
+
+func (m DashboardModel) Update(msg tea.Msg) (tea.Model, tea.Cmd) {
+	var cmd tea.Cmd
+	switch msg := msg.(type) {
+	case tea.KeyMsg:
+		switch msg.String() {
+		case "right", "l":
+			m.tableView = (m.tableView + 1) % 2
+		case "left", "h":
+			if m.tableView-1 <= 0 {
+				m.tableView = 0
+			} else {
+				m.tableView--
+			}
+		case "q", "ctrl+c":
```

Comment:
People might also hit `<ESC>` to quit. And probably `ctrl-d` (which you see alot in interpreters etc)